### PR TITLE
Removed miopengemm and tensile

### PR DESCRIPTION
**Pull request by DrizztDoUrden**
Implements #2217

### CODE REVIEW

**Comment by DrizztDoUrden** [comment]
I don't know who else to add, this is trivial even though tedious

**Review by junliume — APPROVED** [review]

**Comment by junliume** [comment]
Ping @JehandadKhan  before merging.

**Comment by DrizztDoUrden** [comment]
Ping @JehandadKhan

File: src/gemm_v2.cpp [hunk 1]
```diff
@@ -177,55 +169,24 @@ static GemmBackend_t enforce_gemm_backend(miopenDataType_t data_type,
     GemmBackend_t gemm_backend_env      = GemmBackend_t::nogemmbackend;
 
     // enforce backend based on env variable
+    // I have left the commented lines here to preserve values for the enforce and hint at why are
+    // they 1 and 3
```
Comment (JehandadKhan) [inline src/gemm_v2.cpp:173]: I was hoping we would be able to get rid of this function.

File: src/include/miopen/gemm_v2.hpp [hunk 2]
```diff
@@ -38,8 +38,6 @@ enum GemmBackend_t
 {
     nogemmbackend = 0,
     rocblas       = 1,
-    miopengemm    = 2,
-    miopentensile = 3,
```
Comment (JehandadKhan) [inline src/include/miopen/gemm_v2.hpp:42]: This struct is redundant now!

**Review by JehandadKhan — APPROVED** [review]

**Comment by JehandadKhan** [comment]
@DrizztDoUrden  The Jenkinsfile and Dockerfile might need to be updated as well, however, they are minor changes.

**Comment by atamazov** [comment]
@JehandadKhan @junliume 🟠 I heard that we will most likely have to use something other than rocblas for one of the new GPUs. Therefore, I highly recommend keeping the existing infrastructure that allows us to switch between different GEMM backends, so as not to reinvent it later.

File: src/include/miopen/gemm_v2.hpp (hunk 2 above)
Comment (atamazov) [reply src/include/miopen/gemm_v2.hpp:42]: @JehandadKhan maybe not, please see https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2265#issuecomment-1695963969

**Comment by atamazov** [comment]
@DrizztDoUrden Please notice https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2265#issuecomment-1695963969

**Pull request merged**